addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionScanRuleUnitTest.java

```diff
@@ -79,4 +92,76 @@ void shouldReturnExpectedMappings() {
                 tags.get(CommonAlertTag.WSTG_V42_INPV_05_SQLI.getTag()),
                 is(equalTo(CommonAlertTag.WSTG_V42_INPV_05_SQLI.getValue())));
     }
+
+    @Test
+    void shouldDetectTimeBasedInjection() throws HttpMalformedHeaderException {
+        // Given
+        Pattern sleepPattern = Pattern.compile("0\\s+\\|\\|\\s+sleep\\((\\d+)(?:\\.\\d+)?\\)");
+        String regularContent = "<!DOCTYPE html><html><body>Nothing to see here.</body></html>";
+        Set<String> payloadSet = new HashSet<>();
+        nano.addHandler(
+                new NanoServerHandler("/") {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        String value = getFirstParamValue(session, "p");
+                        if (value == null) {
+                            return newFixedLengthResponse(regularContent);
+                        }
+                        if (value.contains("sleep(")) {
+                            payloadSet.add(value);
+                        }
+                        Matcher match = sleepPattern.matcher(value);
+                        if (!match.find()) {
+                            return newFixedLengthResponse(regularContent);
+                        }
+                        try {
+                            int sleepInput = Integer.parseInt(match.group(1));
+                            Thread.sleep(sleepInput * 1000L);
+                        } catch (InterruptedException ex) {
+                            fail("failed to sleep thread for time-based nosql injection");
```

Comment:
`// Ignore` instead, fail here will not fail the test, just the request (in any case the thread will not be interrupted normally).